src/components/shared/Navigation.tsx

```diff
@@ -653,10 +664,35 @@ class NavigationView extends React.Component<Props> {
     routeToPath(params["path"], this.navigation, params);
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     this._unsubscribeFromBranch = branch.subscribe(this._processBranchDeeplink);
+    this._unsubscribeFromNotificationOpen = firebase
+      .notifications()
+      .onNotificationOpened(this.handleNotificationOpen);
+    this.handleNotificationOpen(
+      await firebase.notifications().getInitialNotification()
+    );
   }
 
+  /**
+   * Takes an action based on a notification being opened.
+   */
+  handleNotificationOpen = async (
+    notificationOpen: NotificationOpen | undefined
```

Comment:
What's the different between | undefined vs ?

Reply:
(i think omar's comment was intended to be a reply to this).

Yep - the type is the same, but using `?` allows you to call the method without any arguments, while this forces you to always provide the argument, even if it's `undefined`.